opentelemetry-api/src/opentelemetry/metrics/meter.py

```diff
@@ -0,0 +1,216 @@
+        self, name, callback, unit=None, description=None
+    ) -> ObservableGauge:
+        pass
+
+    @abstractmethod
+    def create_observable_up_down_counter(
+        self, name, callback, unit=None, description=None
+    ) -> ObservableUpDownCounter:
+        pass
+
+    @staticmethod
+    def check_unique_name(checker):
+        def wrapper_0(method):
+            @wraps(method)
+            def wrapper_1(self, name, unit=None, description=None):
+                checker(self, name)
+                return method(self, name, unit=unit, description=description)
+
+            return wrapper_1
+
+        return wrapper_0
+
+
+class DefaultMeter(Meter):
+    def __init__(self):
+        self._instrument_names = set()
+
+    def _instrument_name_checker(self, name):
+
+        if name in self._instrument_names:
+            raise Exception("Instrument name {} has been used already")
+
+        self._instrument_names.add(name)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_counter(self, name, unit=None, description=None) -> Counter:
+        return DefaultCounter(name, unit=unit, description=description)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_up_down_counter(
+        self, name, unit=None, description=None
+    ) -> UpDownCounter:
+        return DefaultUpDownCounter(name, unit=unit, description=description)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_observable_counter(
+        self, name, callback, unit=None, description=None
+    ) -> ObservableCounter:
+        return DefaultObservableCounter(
+            name,
+            callback,
+            unit=unit,
+            description=description,
+        )
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_histogram(self, name, unit=None, description=None) -> Histogram:
+        return DefaultHistogram(name, unit=unit, description=description)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_observable_gauge(
+        self, name, callback, unit=None, description=None
+    ) -> ObservableGauge:
+        return DefaultObservableGauge(  # pylint: disable=abstract-class-instantiated
+            name,
+            callback,
+            unit=unit,
+            description=description,
+        )
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_observable_up_down_counter(
+        self, name, callback, unit=None, description=None
+    ) -> ObservableUpDownCounter:
+        return DefaultObservableUpDownCounter(  # pylint: disable=abstract-class-instantiated
+            name,
+            callback,
+            unit=unit,
+            description=description,
+        )
+
+
+class MeterProvider(ABC):
+    """
+    var
+    """
+
+    @abstractmethod
+    def get_meter(
+        self,
+        name,
+        version=None,
+        schema_url=None,
+    ) -> Meter:
+        """
+        vpas
+        """
+        pass
```

Comment:
Does the `MeterProvider` need a method to add a measurement processor? Similar to `add_span_processor` in the TracerProvider?

Reply:
I believe that is sdk concern. SDK will implement this and adds the extra necessary methods for processing/aggregating/configuring.

Reply:
Right, as @lonewolf3739 commented, this is a SDK thing. :v: